tests/utils/test_async_logging_queue.py

```diff
@@ -70,15 +84,16 @@ def test_single_thread_publish_consume_queue(monkeypatch):
         async_logging_queue.flush()
         # 2 batches are sent to the worker thread pool due to grouping, otherwise it would be 5.
         assert mock_worker_threadpool.submit.call_count == 2
-        assert async_logging_queue.is_idle()
+        assert async_logging_queue.is_active()
         assert mock_check_threadpool.shutdown.call_count == 1
         assert mock_worker_threadpool.shutdown.call_count == 1
 
 
-def test_grouping_batch_in_time_window():
+@pytest.mark.parametrize("async_logging_data", [None], indirect=True)
```

Comment:
I'd use a context manger if I had to do this.

Reply:
I agree with Haru, the current code is cool but kinda hard to read